yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/MultiServerConnectionFactory.java

```diff
@@ -182,103 +174,35 @@ public RelationalPreparedStatement prepareStatement(String sql) throws SQLExcept
             return getCurrentConnection(true, "prepareStatement").prepareStatement(sql);
         }
 
+        @Nullable
         @Override
-        public void setAutoCommit(boolean autoCommit) throws SQLException {
-            if (!autoCommit) {
-                throw new UnsupportedOperationException("setAutoCommit(false) is not supported in YAML tests");
-            }
-            logger.info("Sending operation {} to all connections", "setAutoCommit");
-            for (RelationalConnection connection: relationalConnections) {
-                connection.setAutoCommit(autoCommit);
-            }
-        }
-
-        @Override
-        public boolean getAutoCommit() throws SQLException {
-            return getCurrentConnection(false, "getAutoCommit").getAutoCommit();
+        public MetricCollector getMetricCollector() {
+            throw new UnsupportedOperationException("MultiServer does not support getting the metric collector");
         }
 
+        @Nullable
         @Override
-        public void commit() throws SQLException {
-            throw new UnsupportedOperationException("commit is not supported in YAML tests");
+        public EmbeddedRelationalConnection tryGetEmbedded() {
+            return null;
         }
 
+        @Nonnull
         @Override
-        public void rollback() throws SQLException {
-            throw new UnsupportedOperationException("rollback is not supported in YAML tests");
+        public List<String> getVersions() {
+            return this.versions;
```

Comment:
```suggestion
return getCurrentConnection(false, "getVersion").getVersion();
```

If done this way, maybe we can turn `getVersions` to getVersion and no longer need the list of versions in the multi-server connection?

Reply:
I think the most important thing for logging purposes is that the user is able to reconstruct what happened from the log they give. Getting a list of versions is probably not that helpful without context like where the query started, what version it failed against, etc. 

For my change, I ended up merging your branch in, and then I needed to add an additional method, `initialVersion`, so that it was clear from a connection what version it would start on. That ended up being the clearest way to write assertions on the version, I think. Technically, you could almost rely on the fact that the connection shouldn't have been used yet and so getting the current connection (without advancing) is enough to return the right value...but it seemed a bit shakier and also less clearly associated with the goals it was trying to accomplish than adding a separate method to say where the query started.

Reply:
Sounds like two things are required to make this useful: The initial version (so that assertions can be made for Alec's change) and also the actual history of what transpired during the life of the connection (for reproduction purposes, per Scott's comment). It does look like they are both quite different and hard to combine into one method.

Reply:
The list as it is should provide the `initialVersion` (i.e. `getVersions().get(0)` should always work).

It is potentially less clear what happened. I think it should be clear, because I think you'll have the specific result that failed, so you should know whether it was the first or second continuation.

Reply:
That last comment was written without seeing @ohadzeliger's two comments that came before it.

Reply:
I think, based on my testing, the current list _works_ although it could probably be better, by having multiple different methods to support past vs future versions.
Perhaps worth not trying to iterate on the behavior here until we have more experience working with this in actual investigations.

Reply:
Sure.